src/wp-includes/ms-functions.php

```diff
@@ -2104,7 +2104,7 @@ function wpmu_log_new_registrations( $blog_id, $user_id ) {
 			$wpdb->registration_log,
 			array(
 				'email'           => $user->user_email,
-				'IP'              => preg_replace( '/[^0-9., ]/', '', wp_unslash( $_SERVER['REMOTE_ADDR'] ) ),
+				'IP'              => preg_replace( '/[^a-f0-9:., ]/', '', wp_unslash( $_SERVER['REMOTE_ADDR'] ) ),
```

Comment:
I note that this included `,` and spaces before, which I don't think should be here. REMOTE_ADDR should be a strict IP address only, not commas/spaces which would likely be present in a `X_HTTP_FORWARDED_FOR` header.

Suggest that this should be removed and stricter IP sanitization used:
```suggestion
				'IP'              => preg_replace( '/[^a-f0-9:.]/', '', wp_unslash( $_SERVER['REMOTE_ADDR'] ) ),
```

Reply:
I agree as all references I could find only talk about a single IP address.
I also couldn't find anything about casing though, so maybe uppercase `A-F` should be allowed as well, as lowercase is probably not guaranteed for all servers populating `REMOTE_ADDR`?

Reply:
It dates back to 2006 --

https://mu.trac.wordpress.org/ticket/77
https://mu.trac.wordpress.org/changeset/667

Reply:
(also, we're way past it being reasonable to do, but I'm a bit sad we're storing the IPs in varchars instead of using [INET6_ATON](https://dev.mysql.com/doc/refman/8.4/en/miscellaneous-functions.html#function_inet6-aton) and [`inet_pton()`](https://www.php.net/manual/en/function.inet-pton.php) -- hindsight being 20/20 and all that, and probably wasn't reasonable in 2006 anyways)